mlflow/utils/download_cloud_file_chunk.py

```diff
@@ -0,0 +1,192 @@
+"""
+This script should be executed in a fresh python interpreter process using `subprocess`.
+"""
+import argparse
+import json
+import os
+import requests
+import sys
+import urllib3
+
+from functools import lru_cache
+from packaging.version import Version
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+
+def download_chunk(range_start, range_end, headers, download_path, http_uri):
+    combined_headers = {**headers, "Range": f"bytes={range_start}-{range_end}"}
+
+    with cloud_storage_http_request(
+        "get", http_uri, stream=False, headers=combined_headers
+    ) as response:
+        # File will have been created upstream. Use r+b to ensure chunks
+        # don't overwrite the entire file.
+        with open(download_path, "r+b") as f:
+            f.seek(range_start)
+            f.write(response.content)
+
+
+_TRANSIENT_FAILURE_RESPONSE_CODES = frozenset(
+    [
+        408,  # Request Timeout
+        429,  # Too Many Requests
+        500,  # Internal Server Error
+        502,  # Bad Gateway
+        503,  # Service Unavailable
+        504,  # Gateway Timeout
+    ]
+)
+
+
+@lru_cache(maxsize=64)
+def _cached_get_request_session(
+    max_retries,
+    backoff_factor,
+    retry_codes,
+    # To create a new Session object for each process, we use the process id as the cache key.
+    # This is to avoid sharing the same Session object across processes, which can lead to issues
+    # such as https://stackoverflow.com/q/3724900.
+    _pid,
+):
+    """
+    This function should not be called directly. Instead, use `_get_request_session` below.
+    """
+    assert 0 <= max_retries < 10
+    assert 0 <= backoff_factor < 120
+
+    retry_kwargs = {
+        "total": max_retries,
+        "connect": max_retries,
+        "read": max_retries,
+        "redirect": max_retries,
+        "status": max_retries,
+        "status_forcelist": retry_codes,
+        "backoff_factor": backoff_factor,
+    }
+    if Version(urllib3.__version__) >= Version("1.26.0"):
+        retry_kwargs["allowed_methods"] = None
+    else:
+        retry_kwargs["method_whitelist"] = None
+
+    retry = Retry(**retry_kwargs)
+    adapter = HTTPAdapter(max_retries=retry)
+    session = requests.Session()
+    session.mount("https://", adapter)
+    session.mount("http://", adapter)
+    return session
```

Comment:
This method, along with a bunch of others, is defined in `rest_utils.py`. **However**, if we import these utils from MLflow, it will take ~ 1.5 seconds for the fresh subprocess to import all of MLflow. For prototyping purposes, I redefined these utils so that this module doesn't rely on any imports from MLflow, but we probably want to make this cleaner before merging.

In particular, this creates the potential for Databricks multi-part download logic to diverge from the rest of MLflow.

At minimum, before merging, we need some test verifying that this script doesn't import MLflow and runs quickly.